gcloud/translate/client.py

```diff
@@ -0,0 +1,223 @@
+                            :data:`ENGLISH_ISO_639`.)
+    """
+
+    def __init__(self, key, http=None, target_language=ENGLISH_ISO_639):
+        self.key = key
+        if http is None:
+            http = httplib2.Http()
+        self.connection = Connection(http=http)
+        self.target_language = target_language
+
+    def get_languages(self, target_language=None):
+        """Get list of supported languages for translation.
+
+        Response
+
+        See: https://cloud.google.com/translate/v2/\
+        discovering-supported-languages-with-rest
+
+        :type target_language: str
+        :param target_language: (Optional) The language used to localize
+                                returned language names. Defaults to the
+                                target language on the current client.
+
+        :rtype: list
+        :returns: List of dictionaries. Each dictionary contains a supported
+                  ISO 639-1 language code (using the dictionary key
+                  ``language``). If ``target_language`` is passed, each
+                  dictionary will also contain the name of each supported
+                  language (localized to the target language).
+        """
+        query_params = {'key': self.key}
+        if target_language is None:
+            target_language = self.target_language
+        if target_language is not None:
+            query_params['target'] = target_language
+        response = self.connection.api_request(
+            method='GET', path='/languages', query_params=query_params)
+        return response.get('data', {}).get('languages', ())
+
+    def detect_language(self, values):
+        """Detect the language of a string or list of strings.
+
+        See: https://cloud.google.com/translate/v2/\
+        detecting-language-with-rest
+
+        :type values: str or list
+        :param values: String or list of strings that will have
+                       language detected.
+
+        :rtype: str or list
+        :returns: A list of dictionaries for each queried value. Each
+                  dictionary typically contains three keys (though not
+                  all will be present in all cases)
+
+                  * ``confidence``: The confidence in language detection, a
+                    float between 0 and 1.
+                  * ``input``: The corresponding input value.
+                  * ``language``: The detected language (as an ISO 639-1
+                    language code).
+
+                  If only a single value is passed, then only a single
+                  dictionary will be returned.
+        :raises: :class:`ValueError <exceptions.ValueError>` if the number of
+                 detections is not equal to the number of values.
+                 :class:`ValueError <exceptions.ValueError>` if a value
+                 produces a list of detections with 0 or multiple results
+                 in it.
+        """
+        single_value = False
```

Comment:
1. No I don't think it's **easier** to read, but that's just, like, my opinion man.
1. I'm happy to change it if it's important to you.